src/dafny/ssz/BitListSeDes.dfy

```diff
@@ -69,9 +69,10 @@ include "Constants.dfy"
      *  The algorithm to encode list of bits works as follows:
      *  1. given a list of bits l, 
      *  2. append 1 to l, this is a sentinnelle, and let l' = l + [1] 
-     *  3. if |l'| * 8 is not 0, append 8 - (|l| + 1) % 8 zeros to l' 
-     *     to obtain a list of size multiplw of 8
-     *     let l'' = l' + possibly some [0]
+     *  3. if |l'| % 8 is not 0, append 8 - (|l'| % 8) zeros to l' 
+     *     to obtain a list of size multiple of 8
+     *     let l'' = l', if |l'| % 8 == 0
+     *               l' + [0] * (8 - (|l'| % 8)), if |l'| % 8 != 0
      *     This ensures that |l''| % 8 == 0 and can be seen as a sequence of Bytes
```

Comment:
I am not sure this is true. In BitList we add a `true` bit as a sentinel, so we append 1 zero bit less compared to but vectors.
The code fo this function reflects that fact.
Can you check that again?

Reply:
That should be what is expressed by the comment as we have defined `l' = l + [1]`.

Reply:
The only real fix was changing `|l'| * 8 ` to `|l'| % 8`.
The rest of the changes are supposed to be just minor improvements.